addons/mail/models/discuss/discuss_channel.py

```diff
@@ -437,6 +437,19 @@ def _action_unfollow(self, partner=None, guest=None):
             ],
         )
 
+    def _post_joined_message(self, member=None, partner=None):
```

Comment:
I extracted this part to allow overriding it in livechat, enabling the use of `livechat_username` as the name. This way, we can avoid manually posting the user joining notification and rely on the `add_members` to do so.

Reply:
The other PR targets version 18.0, while this one targets master. Should we fix the issue here and then consider back-porting it to the other PR?

Reply:
Yes we'll backport, whether with 2 PRs or single PR that targets 18.0 it doesn't matter much.

When the PR is likely requiring many changes I prefer target `master` for more flexibility in changes, but here the PR is close to final and it's pretty much ok for 18.0. Thanks!